### PR TITLE
Organize runs page by project

### LINKED ISSUE #467: Make the runs page a data-browsing page
After talking with @arahuja about this, the /runs page should be a way to browse the datasets, projects, and runs we have in a multifaceted way. E.g. we should be able to group runs by dataset, group runs by project, sort + filter them. Maybe CQL on top of the runs page with aggregate/GROUP BY would be sufficient. 

Otherwise, a stopgap measure would be to make a /projects page & a /datasets page.

### PR DESCRIPTION
**Pull request by ihodes**
The projects are sorted so that the project with the most recent run is
at the top, and the runs are sorted by most recent at the top within
each project.

This is the first step to reorganizing the runs page, and one that
requires no changes to the database.

fixes #467 

[Review on Reviewable](https://reviewable.io/reviews/hammerlab/cycledash/472)

### CODE REVIEW

**Comment by ihodes** [comment]
I need to add the dpxdt screenshots, but I'm getting some weird errors I'll fix later. Figured this could go to review before then.

**Comment by danvk** [comment]
@arahuja you cool w/ this UI?

File: cycledash/static/js/runs/components/RunsPage.js [hunk 1]
```diff
@@ -64,9 +57,71 @@ var RunsPage = React.createClass({
...
     });
     var runs = this.filteredRuns();
-    var rows = runs.map(run => {
-      var rows = [<RunRow run={run} key={run.id}
-                          handleClick={this.handleClickRun(run.id)} />];
+    var projectRuns = _.chain(runs)
+        .groupBy('project_name')
+        .map((runs, projectName) => [projectName, runs])
+        .value();
+    var projectTables = _.chain(projectRuns).sortBy(function([projectName, runs]) {
```
Comment (danvk) [inline cycledash/static/js/runs/components/RunsPage.js]: merge this into the `chain`?

File: cycledash/static/js/runs/components/RunsPage.js [hunk 2]
```diff
@@ -64,9 +57,71 @@ var RunsPage = React.createClass({
...
+      return -runs[0].id; // to sort by the descending ID
+    }).map(function([projectName, runs]) {
```
Comment (danvk) [inline cycledash/static/js/runs/components/RunsPage.js]: is there a bug out for the fat-arrow/destructuring clash? it's a shame.

**Comment by danvk** [comment]
Looks fine code-wise

File: cycledash/static/js/runs/components/RunsPage.js (hunk 1 above)
Comment (ihodes) [inline cycledash/static/js/runs/components/RunsPage.js]: Good idea! Done.

**Comment by arahuja** [comment]
Yea this looks good, its the same datafields, but grouped into projects right?

**Comment by ihodes** [comment]
Yeah, simple change for now. Will keep the PRs coming to get to the point we were talking about.

File: cycledash/static/js/runs/components/RunsPage.js (hunk 2 above)
Comment (ihodes) [inline cycledash/static/js/runs/components/RunsPage.js]: I haven't actually looked… may be worth doing. It's definitely a shame.

**Pull request merged**